src/openrct2/network/ServerList.h

```diff
@@ -35,7 +35,12 @@ struct ServerListEntry
     int32_t CompareTo(const ServerListEntry& other) const;
     bool IsVersionValid() const;
 
-    static std::optional<ServerListEntry> FromJson(const json_t* root);
+    /**
+     * Constructs a ServerListEntry from JSON
+     * @param server JSON representing server. Must be object type.
```

Comment:
We're not enforcing that it it an object type on the function, maybe we should

Reply:
There are a few functions like this. With this and with the others, there's always a check further up the callstack that this is an object. I'm not sure whether we should just double up on the checks.